### ISSUE TITLE
Feature request: configurable NoFinalizedLocalVariables

### ISSUE THREAD

**Issue by fprochazka** [openrewrite/rewrite-static-analysis#514]
It would be great to allow configuring `org.openrewrite.java.cleanup.NoFinalizedLocalVariables` to remove `final` only from locally-declared variables, so it ignores method parameters.

I want to prevent writing to variables that are passed in method arguments, and IDEA can generate the `final` there automatically, but it's a lot of noise for local variables that I would like to get rid of.

**Comment by greg-at-moderne** [comment]
Moving over to rewrite-static-analysis because this is where `NoFinalizedLocalVariables` lives now.

**Comment by timtebeek** [comment]
Yes I could see that make sense; elsewhere we've in such cases added a nulable Boolean flag to the recipe, and when `null` kept the old behavior. We'd welcome such a change here as well if you're up for it. Should be relatively straightforward.
